src/lib/SpecialFileDirGen.cc

```diff
@@ -4,6 +4,58 @@
 
 namespace simeng {
 
+// Wrapper around calls to "system(command)". Checks that a shell is available
+// before calling "system" and checking the output for any issues.
+// ensureExitSuccess is used to check for a successful termination status (0)
+// from the child shell, defaults to true
+int systemWrapper(const std::string& command,
+                  const bool ensureExitSuccess = true) {
+  // Check that there is a shell available
+  if (!system(NULL)) {
+    std::cerr
+        << "[SimEng:SpecialFileDirGen] Shell unavailable, can't call system"
+        << std::endl;
+    exit(EXIT_FAILURE);
+  }
+
+  int output = system(command.c_str());
+
+  if (output == -1) {
+    std::cerr << "[SimEng:SpecialFileDirGen] Child process could not be "
+                 "created, or its status could "
+                 "not be retrieved. errno = "
+              << errno << std::endl;
+    exit(EXIT_FAILURE);
+  } else if (WIFEXITED(output) && WEXITSTATUS(output) == 127) {
```

Comment:
Rather than 127, is there an associated ENUM / defined variable? Applied throughout for the rest of these cases

Reply:
The man page specifically states "with the status 127" as well as the other numbers used throughout. I don't think there is an enum

https://man7.org/linux/man-pages/man3/system.3.html